lib/src/log_handler.dart

```diff
@@ -0,0 +1,99 @@
+import 'dart:convert';
+
+import 'package:file/file.dart';
+import 'package:path/path.dart' as p;
+
+import 'constants.dart';
+import 'initializer.dart';
+
+class LogFileStats {
+  /// The oldest timestamp in the log file
+  final DateTime startDateTime;
+
+  /// The latest timestamp in the log file
+  final DateTime endDateTime;
+
+  /// The number of unique session ids found in the log file
+  final int sessionCount;
+
+  /// Contains the data from the [LogHandler.logFileStats]
+  LogFileStats({
```

Comment:
nit make a constructor const if it can be, then analyzer will be able to identify places where a construction can be marked const

```suggestion
  const LogFileStats({
```